test/smoke/run-smoke-suite.sh

```diff
@@ -92,24 +92,74 @@ NODE_UUID=$(kubectl exec "${POD}" -c controller -- cat /opt/wallarm/etc/wallarm/
 echo "UUID: ${NODE_UUID}"
 
 echo "Deploying pytest pod ..."
-kubectl run pytest \
-  --env="NODE_BASE_URL=${NODE_BASE_URL}" \
-  --env="NODE_UUID=${NODE_UUID}" \
-  --env="WALLARM_API_HOST=${WALLARM_API_HOST}" \
-  --env="API_CA_VERIFY=${WALLARM_API_CA_VERIFY}" \
-  --env="CLIENT_ID=${CLIENT_ID}" \
-  --env="USER_UUID=${USER_UUID}" \
-  --env="USER_SECRET=${USER_SECRET}" \
-  --env="HOSTNAME_OLD_NODE=${HOSTNAME_OLD_NODE}" \
-  --image="${SMOKE_IMAGE_NAME}:${SMOKE_IMAGE_TAG}" \
-  --image-pull-policy=IfNotPresent \
-  --pod-running-timeout=3m0s \
-  --restart=Never \
-  --overrides='{"apiVersion": "v1", "spec":{"terminationGracePeriodSeconds": 0, "imagePullSecrets": [{"name": "'"${SMOKE_IMAGE_PULL_SECRET_NAME}"'"}]}}' \
-  --command -- sleep infinity
-
-kubectl wait --for=condition=Ready pods --all --timeout=180s
+
+POD_CONFIG=$(cat << EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pytest
+spec:
+  terminationGracePeriodSeconds: 0
+  restartPolicy: Never
+  imagePullSecrets:
+    - name: "${SMOKE_IMAGE_PULL_SECRET_NAME}"
+  containers:
+  - command:
+    - sleep
+    - infinity
+    env:
+    - name: NODE_BASE_URL
+      value: "${NODE_BASE_URL}"
+    - name: NODE_UUID
+      value: "${NODE_UUID}"
+    - name: WALLARM_API_HOST
+      value: "${WALLARM_API_HOST}"
+    - name: API_CA_VERIFY
+      value: "${WALLARM_API_CA_VERIFY}"
+    - name: CLIENT_ID
+      value: "${CLIENT_ID}"
+    - name: USER_UUID
+      value: "${USER_UUID}"
+    - name: USER_SECRET
+      value: "${USER_SECRET}"
+    - name: HOSTNAME_OLD_NODE
+      value: "${HOSTNAME_OLD_NODE}"
+    image: "${SMOKE_IMAGE_NAME}:${SMOKE_IMAGE_TAG}"
+    imagePullPolicy: IfNotPresent
+    name: pytest
+EOF
+)
+
+if [ "${ALLURE_GENERATE_REPORT:-false}" = "true" ]; then
+POD_CONFIG+=$(cat << EOF
+
+    volumeMounts:
+    - mountPath: /tests/_out/allure_report
+      name: allure-report
+      readOnly: false
+  volumes:
+  - name: allure-report
+    hostPath:
+      path: /allure_report
+      type: DirectoryOrCreate
+EOF
+)
+fi
+
+echo "${POD_CONFIG}" | kubectl apply -f -
+
+
+echo "Getting logs ..."
+kubectl wait --for=condition=Ready pods --all --timeout=600s
 
 echo "Run smoke tests ..."
 trap get_logs_and_fail ERR
-kubectl exec pytest ${EXEC_ARGS} -- pytest -n ${PYTEST_WORKERS} ${PYTEST_ARGS}
\ No newline at end of file
+
+if [ "${ALLURE_UPLOAD_REPORT:-false}" = "true" ]; then
+  RUN_TESTS='allurectl watch -- pytest'
+else
+  RUN_TESTS="pytest"
+fi
+
+echo "Executing command: kubectl exec pytest ${EXEC_ARGS} -- ${RUN_TESTS} -n ${PYTEST_WORKERS} ${PYTEST_ARGS}" && \
+kubectl exec pytest ${EXEC_ARGS} -- ${RUN_TESTS} -n "${PYTEST_WORKERS}" "${PYTEST_ARGS}"
```

Comment:
no quotes here. I know that linter complain)

Reply:
remove everywhere quotes